### ISSUE TITLE
Bad Gateway

### ISSUE THREAD

**Issue by DanSman** [librenms/docker#50]
Hi

I have a problem since the last about 4-5 updates. I didn't have enough time to create an issue straightaway, and I tried to solve it myself first. Which obviously didn't work.
  
I've been using your docker image for, like, six months. At that time I think that it only came with traefik.  
The problem is that I can no longer access the GUI. The only thing I get is `Bad Gateway`.

I use LibreNMS at home on an Arch system.

docker-compose.yml
```
 version: "3.5"

services:
  traefik:
    image: traefik:1.7-alpine
    container_name: traefik
    command:
      - "--logLevel=INFO"
      - "--defaultentrypoints=http,https"
      - "--entryPoints=Name:http Address::8881 Redirect.EntryPoint:https"
      - "--entryPoints=Name:https Address::44300 TLS"
      - "--docker"
      - "--docker.exposedbydefault=false"
      - "--docker.domain=192.168.1.76"
      - "--acme=true"
      - "--acme.acmelogging=true"
      - "--acme.email=admin"
      - "--acme.storage=acme.json"
      - "--acme.entryPoint=https"
      - "--acme.onhostrule=true"
      - "--acme.httpchallenge=true"
      - "--acme.httpchallenge.entrypoint=http"
    ports:
      - target: 8881
        published: 8881
        protocol: tcp
      - target: 44300
        published: 44300
        protocol: tcp
    volumes:
      - "./acme.json:/acme.json"
      - "/var/run/docker.sock:/var/run/docker.sock"
    restart: always

  db:
    image: mariadb:10.2
    container_name: librenms_db
    command:
      - "mysqld"
      - "--sql-mode="
      - "--innodb-file-per-table=1"
      - "--lower-case-table-names=0"
      - "--character-set-server=utf8"
      - "--collation-server=utf8_unicode_ci"
    volumes:
      - "./db:/var/lib/mysql"
    environment:
      - "TZ=${TZ}"
      - "MYSQL_ALLOW_EMPTY_PASSWORD=yes"
      - "MYSQL_DATABASE=${MYSQL_DATABASE}"
      - "MYSQL_USER=${MYSQL_USER}"
      - "MYSQL_PASSWORD=${MYSQL_PASSWORD}"
    restart: always

  memcached:
    image: memcached:alpine
    container_name: librenms_memcached
    environment:
      - "TZ=${TZ}"
    restart: always

  rrdcached:
    image: crazymax/rrdcached
    container_name: librenms_rrdcached
    volumes:
      - "./librenms/rrd:/data/db"
      - "./rrd-journal:/data/journal"
    environment:
      - "TZ=${TZ}"
      - "LOG_LEVEL=LOG_INFO"
      - "WRITE_TIMEOUT=1800"
      - "WRITE_JITTER=1800"
      - "WRITE_THREADS=4"
      - "FLUSH_DEAD_DATA_INTERVAL=3600"
    restart: always

  smtp:
    image: juanluisbaptiste/postfix
    container_name: librenms_smtp
    volumes:
      - "/etc/localtime:/etc/localtime:ro"
    environment:
      - "SERVER_HOSTNAME=192.168.1.76"
      - "SMTP_SERVER=${SMTP_SERVER}"
      - "SMTP_USERNAME=${SMTP_USERNAME}"
      - "SMTP_PASSWORD=${SMTP_PASSWORD}"
    restart: always

  librenms:
    image: librenms/librenms:latest
    container_name: librenms
    domainname: 192.168.1.76
    hostname: librenms
    depends_on:
      - db
      - memcached
      - rrdcached
      - smtp
    volumes:
      - "./librenms:/data"
    labels:
      - "traefik.enable=true"
      - "traefik.backend=librenms"
      - "traefik.port=8881"
      - "traefik.frontend.rule=Host:192.168.1.76"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST=db"
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "DB_TIMEOUT=60"
    env_file:
      - "./librenms.env"
    restart: always
  cron:
    image: librenms/librenms:latest
    container_name: librenms_cron
    domainname: 192.168.1.76
    hostname: librenms
    depends_on:
      - librenms
    volumes:
      - "./librenms:/data"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST=db"
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "DB_TIMEOUT=60"
      - "SIDECAR_CRON=1"
    env_file:
      - "./librenms.env"
    restart: always

  syslog-ng:
    image: librenms/librenms:latest
    container_name: librenms_syslog
    domainname: 192.168.1.76
    hostname: librenms
    depends_on:
      - librenms
    ports:
      - target: 514
        published: 514
        protocol: tcp
      - target: 514
        published: 514
        protocol: udp
    volumes:
      - "./librenms:/data"
    environment:
      - "TZ=${TZ}"
      - "PUID=${PUID}"
      - "PGID=${PGID}"
      - "DB_HOST=db"
      - "DB_NAME=${MYSQL_DATABASE}"
      - "DB_USER=${MYSQL_USER}"
      - "DB_PASSWORD=${MYSQL_PASSWORD}"
      - "DB_TIMEOUT=60"
      - "SIDECAR_SYSLOGNG=1"
    env_file:
      - "./librenms.env"
    restart: always
```
I had to change the ports because they are already in use. The ports in the configuration are open.
  
Maybe I can't see the wood for the trees anymore. I also tried without traefik. Actually both of your [examples](https://github.com/librenms/docker/tree/master/examples).  But nothing.

Maybe you have an idea?

Thank you very much
Dan

**Comment by crazy-max** [comment]
Hi,

Can you post your `librenms.env` file and also `- "traefik.port=8881"` should be `- "traefik.port=8000"`

**Comment by DanSman** [comment]
That was a quick response.  

And you already gave me the hint/solution! Thanks for that!  
`- "traefik.port=8881"` was my problem. Actually, I think I changed it after my problems.  
  
I also checked the nginx configuration in your docker image and librenms runs on port 8000 there.  So, librenms image is passing the `- "traefik.port=8000"` to traefik image. I'm still learning about Docker etc.
  
`librenms.env` was the same as your examples.
  
Anyway, thank you very much for the quick solution.
